library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -971,27 +972,97 @@ private List<ElementSeed> getElementSeeds(final Iterable<Object> ids) {
             // Extract source and destination from ID list
             } else if (id instanceof Iterable) {
                 ((Iterable<?>) id).forEach(edgeIdList::add);
-            // Attempt to extract source and destination IDs from a string form of an array/list
-            } else if ((id instanceof String) && (((String) id).matches("^\\[.*,.*\\]$"))) {
-                edgeIdList = Arrays.asList(((String) id)
-                        .replaceAll("\\s", "")
-                        .replace("[", "")
-                        .replace("]", "")
-                        .split(","));
-            // Assume entity ID as fallback
+            // Attempt to extract source and destination IDs from a string form of an
+            // array/list
+            } else if ((id instanceof String) && (((String) id).matches("^\\[.*,.*\\]$")
+                || ((String) id).matches("^\\[.*,[a-zA-Z0-9|-]*,.*\\]$"))) {
```

Comment:
We could just use one regex for both here
```java
// Matches [src,dest] and [src,label,dest]
EDGE_ID_REGEX = "^\\[(?<src>[a-zA-Z0-9|-]*)(,\\s*(?<label>[a-zA-Z0-9|-]*))?\\s*,\\s*(?<dest>[a-zA-Z0-9|-]*)\\]$";`
```
allowing us to just do something like
```java
if (matcher.matches()) {
  seeds.add(new EdgeSeed(matcher.group("src"), matcher.group("dest")))
}
```
so we can get rid of the `edgeIdList` size checks and the `parseString` method